tf2_geometry_msgs/test/test_tf2_geometry_msgs.cpp

```diff
@@ -74,6 +74,44 @@ TEST(TfGeometry, Frame)
   EXPECT_NEAR(v_advanced.pose.orientation.w, 1.0, EPS);
 }
 
+TEST(TfGeometry, PoseWithCovarianceStamped)
+{
+  geometry_msgs::PoseWithCovarianceStamped v1;
+  v1.pose.pose.position.x = 1;
+  v1.pose.pose.position.y = 2;
+  v1.pose.pose.position.z = 3;
+  v1.pose.pose.orientation.x = 1;
+  v1.header.stamp = ros::Time(2);
+  v1.header.frame_id = "A";
+  v1.pose.covariance[0] = 1;
+  v1.pose.covariance[7] = 1;
+  v1.pose.covariance[13] = 1;
+  v1.pose.covariance[20] = 1;
+  v1.pose.covariance[27] = 1;
+  v1.pose.covariance[35] = 1;
+  
+  // simple api
+  const geometry_msgs::PoseWithCovarianceStamped v_simple = tf_buffer->transform(v1, "B", ros::Duration(2.0));
```

Comment:
Can you add some basic checks of the covariance values? That's a core part of this and there's no coverage of it in these tests.

Reply:
Sure thing.

Reply:
Actually, just realized the transform t used in all these tests performs no rotations. I'll need to declare a local tf_buffer and Transform for my test to do a rotation on the covariance matrix.